scipy/stats/_distn_infrastructure.py

```diff
@@ -2212,21 +2239,36 @@ def fit(self, data, *args, **kwds):
               function to be optimized) and ``disp=0`` to suppress
               output as keyword arguments.
 
+            - method : The method to use. The default is "MLE" (Maximum
+              Likelihood Estimate); "GMM" (Generalized Method of Moments)
+              is also available.
+
+            - order : The number of moments to fit with ``method="GMM"``.
+              This integer must be greater than or equal to the number of
+              non-fixed distribution parameters (including location and scale);
+              the default is the number of non-fixed parameters. This keyword
+              is ignored when ``method="MLE"``.
+
         Returns
         -------
-        mle_tuple : tuple of floats
-            MLEs for any shape parameters (if applicable), followed by those
+        parameter_tuple : tuple of floats
+            Estimates for any shape parameters (if applicable), followed by those
             for location and scale. For most random variables, shape statistics
             will be returned, but there are exceptions (e.g. ``norm``).
 
         Notes
         -----
-        This fit is computed by maximizing a log-likelihood function, with
-        penalty applied for samples outside of range of the distribution. The
-        returned answer is not guaranteed to be the globally optimal MLE, it
+        With ``method="MLE"`` (default),
+        the fit is computed by maximizing a log-likelihood function, with
+        penalty applied for samples beyond the range of the distribution.
+
+        With ``method="GMM"``, the fit is computed by minimizing a weighted
```

Comment:
will there be a method to define `callable`s? Not super familiar with GMM method in general but from reading: 
https://en.wikipedia.org/wiki/Generalized_method_of_moments
it would seem that users may prefer to pass in their own custom moment functions (e.g. `g(y, theta)` from wikipedia page. A sensible  default here would be to take the first `k` moments, for the `k` parameters, assuming the moments are all finite.

Reply:
I didn't plan on that. Is this possible for MLE? This is a method of `rv_continuous` classes, so I suppose the user can do that by creating their own distribution class (e.g. subclassing an existing one) and defining the `moment` method.

Reply:
> I didn't plan on that. Is this possible for MLE? 

It is not possible for `MLE` b/c the ***choice*** of MLE method defines the function to be the score (first derivative of the log likelihood). For the `MME` method there are expectations to be calculated, e.g. `E(g(x, theta))` where `x~f(x,theta)`. There would seem to be some sensible defaults (e.g. `x^k` for `k=1,2` etc. as needed) but in some cases these will be undefined or infinite (Cauchy) but choosing a different `g(x, theta)` would make the expectations under the  distribution finite (e.g. in the Cauchy case, take `g = (1+x^2)^{-k}` for `k>=1` makes the first moment finite, similar for higher moments. 
It would seems like that is already going to happen with the `moment` method being the callable passed in no matter what, right? 

I get that there is a time constraint, so can you somehow leave open the option for passing a custom `callable` function `g` in the future if it seems like you won't be able to complete that part for first implementation?

Reply:
I think this is addressed by de-scoping the "generalized" part of this PR. Re-reading about generalized method of moments, I see that passing in callables could be important. Originally, I was going to have `fit` accept an "order" parameter, but to make it more general, that could just as well have been a list of callables or a single callable and list of parameters.

Reply:
>I think this is addressed by de-scoping the "generalized" part of this PR.

I think so, I prefer the new approach for the first pass, marking as resolved, thanks